genmc-sys/cpp/src/MiriInterface/Mutex.cpp

```diff
@@ -0,0 +1,142 @@
+/** This file contains functionality related to handling mutexes.  */
+
+#include "MiriInterface.hpp"
+
+// CXX.rs generated headers:
+#include "genmc-sys/src/lib.rs.h"
+
+auto MiriGenmcShim::handle_mutex_lock(ThreadId thread_id, uint64_t address, uint64_t size)
+    -> MutexLockResult {
+    // FIXME(genmc,multithreading): ensure this annotation id is unique even if Miri runs GenMC mode
+    // multithreaded. We cannot use the address directly, since it is 64 bits, and `ID` is an
+    // `unsigned int`.
+    ModuleID::ID annot_id;
+    auto [it, inserted] = annotation_id.try_emplace(address, annotation_id_counter);
+    if (inserted) {
+        annot_id = annotation_id_counter++;
+    } else {
+        annot_id = it->second;
+    }
+    const auto annot = std::move(Annotation(
+        AssumeType::Spinloop,
+        Annotation::ExprVP(NeExpr<AnnotID>::create(
+                               RegisterExpr<AnnotID>::create(ASize(size).getBits(), annot_id),
+                               ConcreteExpr<AnnotID>::create(ASize(size).getBits(), SVal(1))
+        )
+                               .release())
```

Comment:
What is all of this about? Why does `ASize(size).getBits()` show up twice and what does the `1` mean?